src/freertos_drivers/common/WiFiInterface.hxx

```diff
@@ -0,0 +1,377 @@
+    /// @param security_type specifies security type.
+    /// @param timeout is the number of milliseconds to allow for a connection
+    /// to the access point. A value of zero indicates block until a final
+    /// status has been reached (successful or otherwise).
+    ///
+    /// @return @ref WiFiErrorCode as the result of the connection attempt.
+    virtual WiFiErrorCode connect(const std::string &ssid,
+                                  const std::string &password,
+                                  WiFiSecurity security_type,
+                                  const uint32_t timeout = 0) = 0;
+
+    /// Connects to an access point based on stored SSID profiles.
+    ///
+    /// The SSID profiles will be ordered first by priority and second by the
+    /// RSSI value for the access point. Note that a successful connection is
+    /// not guaranteed by this method.
+    ///
+    /// Note: This API is synchronous and will block until the timeout value or
+    /// the SSID connection completes.
+    ///
+    /// @param timeout is the number of milliseconds to allow for a connection
+    /// to the access point. A value of zero indicates block until a final
+    /// status has been reached (successful or otherwise).
+    ///
+    /// @return @ref WiFiErrorCode as the result of the connection attempt.
+    virtual WiFiErrorCode connect(const uint32_t timeout = 0) = 0;
+
+    /// Creates an access point on this WiFi interface.
+    ///
+    /// Note: The ap_ip and dns_ip are in NETWORK byte order (big endian).
+    ///
+    /// @param ssid access point ssid.
+    /// @param password password of the access point (preshared key), a blank
+    /// string is only valid for @ref WiFiSecurity::OPEN.
+    /// @param security specifies security type.
+    /// @param ap_ip is the IP address for the access point.
+    /// @param dns_ip is the DNS server IP address for the access point to
+    /// provide to stations.
+    ///
+    /// @return @ref WiFiErrorCode as the result of setting up the access point.
+    virtual WiFiErrorCode setup_access_point(const std::string &ssid,
+                                             const std::string &password = "",
+                                             WiFiSecurity security = WiFiSecurity::OPEN,
+                                             uint32_t ap_ip = 0,
+                                             uint32_t dns_ip = 0) = 0;
+
+    /// Initiate the WPS Push Button Control connection process.
+    ///
+    /// @param blocking will cause this method to block until the WPS process
+    /// completes. Default is non-blocking.
+    /// @param timeout is the number of milliseconds to wait for the WPS
+    /// process to complete. A value of zero indicates that the interface
+    /// default timeout should be used.
+    ///
+    /// @return @ref WiFiErrorCode as the result of the connection attempt.
+    virtual WiFiErrorCode initiate_wps_connect(bool blocking = false,
```

Comment:
WPS has multiple modes of operation.  For example, there is push button connect (PBC) as well as PIN mode.  This is why the CC32xxWiFi::wlan_wps_pbc_initiate() with "pbc" suffix was used.  Consider adding the "pbc" suffix.